### DISCUSSION TITLE
Connection Problem with devices when using an USB hub

### DISCUSSION THREAD

**Discussion by spamin47** [appium/appium#20729] [Q&A]
The problem
Hello, I have connections issues with my Android devices with Adb when I use a USB hub

Devices are disconnected regulary when I use an autoalimented hub (I use [this hub](https://www.amazon.fr/Plugable-SuperSpeed-adaptateur-secteur-appareils/dp/B00RNFY8HQ/ref=sr_1_fkmr0_1?ie=UTF8&qid=1542271619&sr=8-1-fkmr0&keywords=Plugable+USB+2.0+7-Port+High+Speed+Charging+Hub+with+60W+Power+Adapter) who is advice by [openstf](https://github.com/openstf/stf) ). I already try others hub, and i havec the same problem :'(

I don't have any problem when I plug devices directly on Computer Machine.

Have you any idea to resolve the problem ?

Environment
Appium version that exhibits the issue: 2.10.3
Last Appium version that did not exhibit the issue (if applicable): 1.8.1
Desktop OS/version used to run Appium: Ubuntu 22.04.5
Node.js version (unless using Appium.app|exe): v16.20.0
Mobile platform/version under test: Android
Real device or emulator/simulator: Real device throw a USB hub
Appium CLI or Appium.app|exe: CLI

**Comment by portyu9** [comment]
Hi @spamin47 

This is not an Appium issue. If `adb` itself loses the devices only when they are connected through the USB hub, Appium has already lost its underlying device transport.

Watch the USB/ADB connection directly:

```bash
adb track-devices
```

and in another terminal:

```bash
sudo dmesg -w
```

If you see USB resets/disconnects, fix the USB layer:

* use a genuinely powered hub with enough power per device
* use short, high-quality data cables
* split many devices across multiple USB controllers/hubs
* disable Linux USB autosuspend for those devices/hubs if it is suspending them

You can also confirm the diagnosis by running the devices for a while through the hub with **Appium completely stopped**. If `adb devices` still drops them, Appium cannot fix it.

Since direct USB connections are stable, changing Appium capabilities or drivers is not the solution—the hub/power/USB transport must be stabilized.

Please mark this answer as accepted if it helped, thank you.
